### DISCUSSION TITLE
NextJS local images vs CDN images

### DISCUSSION THREAD

**Discussion by sud0bug** [vercel/community#1089] [Help] [answered]
What is the ideal way to host images inside a NextJS repo? Is it better to serve from a CDN link or put images directly in local and import and use it.?

**Comment by swarnava** [comment, marked as answer]
Hi, there is no recommended way of doing it. It's all about Developer personal preference. Vercel also provide Image optimization which you can leverage if you host in Vercel: https://vercel.com/docs/concepts/image-optimization

**Reply by sud0bug** [reply]
Ok. I was trying to understand the better option from a performance perspective. I saw in our org some developers use CDN while others put images in local. Wanted to setup a uniform approach here.

**Reply by amyegan** [reply]
The [image optimization](https://vercel.com/docs/concepts/image-optimization) feature that Swarnava mentioned is a good reason to host images with Vercel. It automates a lot of optimization tasks, like serving different image sizes based on the device's screen size, showing a blur-up placeholder while an image is loading in the Next image component, and using lazy-loading and Edge caching to improve initial page load performance. You can also take advantage of image optimization features with [remote images](https://vercel.com/docs/concepts/image-optimization/remote-images) from a CDN with a little extra configuration. 

But if your site has a large number of images, or very large images in general, then you may run into [image optimization limits](https://vercel.com/docs/concepts/image-optimization/limits-and-pricing) and moving images entirely to a separate CDN could be a better option in that case. Choosing the best solution depends on your site and your preferred way of working.
